torch/_prims/__init__.py

```diff
@@ -259,22 +238,13 @@ def TensorMeta(
     if isinstance(device, str):
         device = torch.device(device)
 
-    if isinstance(tensorlike, FakeTensor):
-        mode = tensorlike.fake_mode
-    else:
-        mode = get_prim_fake_mode()
-
-    if device.type == "meta":
-        return torch.empty_strided(shape, strides, dtype=dtype, device="meta")
+    # SymInt doesnt support empty_strided yet
+    if any(
+        isinstance(inp, torch.SymIntNode) for inp in itertools.chain(shape, strides)
+    ):
+        return torch.empty(shape, dtype, device=device)
```

Comment:
Should this just unconditionally call `empty_strided` once it has symint support?

I can try to update it once https://github.com/pytorch/pytorch/pull/81238/files lands (I'm updating it today)